client/app/(auth)/createProfile.tsx

```diff
@@ -1,19 +1,430 @@
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedAvatarOption, setSelectedAvatarOption] =
+    useState<AvatarOption>(avatarOptions[0]);
+  const [showBottomDrawer, setShowBottomDrawer] = useState(false);
+  const [showDatePicker, setShowDatePicker] = useState(false);
+  const [selectedDate, setSelectedDate] = useState<Date>();
+
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm<UserEditModel>({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      avatarUrl: "",
+      dateOfBirth: undefined,
+      username: "",
+      name: "",
+    },
+    mode: "onSubmit",
+  });
+
+  useEffect(() => {
+    const setDefaultUserData = async () => {
+      if (currentUser) {
+        const userData = await UserController.getUser(currentUser.uid);
+        if (userData) {
+          const matchedAvatar = avatarOptions.find(
+            (option) => option.url === userData!.avatarUrl,
+          );
+          if (matchedAvatar) {
+            setSelectedAvatarOption(matchedAvatar);
+          }
+
+          reset({
+            avatarUrl: userData.avatarUrl,
+            dateOfBirth: userData.dateOfBirth
+              ? new Date(userData.dateOfBirth)
+              : undefined,
+            name: userData.name,
+            username: userData.username,
+          });
+        }
+      }
+    };
+
+    setDefaultUserData();
+  }, [currentUser, reset]);
+
+  const handleCreateProfile = async (model: UserEditModel) => {
+    setIsLoading(true);
+    try {
+      if (currentUser) {
+        await UserController.editUser(currentUser.uid, model);
+        const userData = await UserController.getUser(currentUser.uid);
+        dispatch(setUser(userData));
+        router.push("/(auth)/createTeam");
+      } else {
+        router.push("/(auth)");
+      }
+    } catch (error) {
+      console.log(error);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
   return (
-    <View className="flex-1 items-center justify-center p-5">
-      <Text className="pbk-h6 text-gray-950">This is temporary</Text>
-      <Text className="pbk-b1 text-gray-950">{user.userId}</Text>
-      <Text className="pbk-b1 text-gray-950">{user.email}</Text>
-      <Text className="pbk-b1 text-gray-950">{user.username}</Text>
-
-      <Link className="mt-4 py-4" href="/">
-        <Text className="pbk-b2 text-purple-500">Go to home screen!</Text>
-      </Link>
+    <View className="flex-1 bg-gray-950">
+      <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
+        <View className="flex-1">
+          <KeyboardAvoidingView behavior="padding" className="flex-1">
+            <View className="flex-1 flex-col px-6 py-4">
+              <View className="my-8 size-8 self-end" />
+
+              <Text className="pbk-h5 mb-8 text-base-white">
+                Tell us who you are
+              </Text>
+
+              <Text className="pbk-b2 mb-1.5 text-base-white">Username</Text>
```

Comment:
I recently made this change in my PR, and it still needs a final review, but what do you think about moving this label into the Controller? I think that makes it a bit more readable as it ties the label to the TextInput. Note: You would need to make this change on all controllers if you agree

```suggestion
              <Text className="pbk-b2 mb-1.5 text-base-white">Username</Text>
              <Controller 
                  ....
                  render= {...} => {
                  <>
                      <Text className="pbk-b2 mb-1.5 text-base-white">Username</Text>
                      <View
                          className={`mb-2 min-h-14 w-full flex-row items-center rounded-xl border             
                          ${errors.username ? "border-red-600" : "border-gray-920"} px-2 py-2`}
                       ></View>
                  </>
```

Reply:
Yeah I think that's a good idea. I changed it for all the other for Controllers as well now. I think in the future we can eventually just wrap the `Controller` component and pass in a `label` prop, where it is automatically set in the `Text` component